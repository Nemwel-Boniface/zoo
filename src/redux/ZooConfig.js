import { createStore,  applyMiddleware } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import ZooReducer from "./Zoo/Zoo"

const middleWares = [thunk, logger ]

const store = createStore(
  ZooReducer,
  applyMiddleware(...middleWares)
)
export default store;