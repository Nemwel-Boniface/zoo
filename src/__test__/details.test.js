import { createStore, applyMiddleware, combineReducers } from 'redux';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import ZooReducer from '../redux/Zoo/Zoo';
import Details from '../components/details';

const zoored = combineReducers({ animals: ZooReducer });
const middleWares = [thunk, logger];

const store = createStore(
  zoored,
  applyMiddleware(...middleWares),
);

afterEach(cleanup);

describe('Test for the details Page', () => {
  test('Check whether the details page renders correctly', () => {
    const detailsPage = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Details />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(detailsPage).toMatchSnapshot();
  });
});

describe("Test for each animal's details card", () => {
  test('Check whether the animals details Card renders correctly', () => {
    const animalsDetailsCard = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Details
              id={124}
              name="Nigerian Dwarf Goat"
              image="https://upload.wikimedia.org/wikipedia/commons/f/fc/Nigerdwarfgoat.jpg"
              animalType="Mammal"
            />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(animalsDetailsCard).toMatchSnapshot();
  });
});
