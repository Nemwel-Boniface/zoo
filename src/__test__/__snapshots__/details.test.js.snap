// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for each animal's details card Check whether the animals details Card renders correctly 1`] = `
<div
  className="Details"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-arrow-left"
    />
    Home
  </button>
  <div
    className="animalDetails"
  >
    <h2 />
    <img
      alt="undefined details"
    />
    <h2
      className="characteristics"
    >
      Characteristics
    </h2>
    <p
      className="padded"
    >
      Latin Name
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Animal Type
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Active Times
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Minimum Length
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Maximum Length
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Minimum Weight
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Maximum Weight
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Life span
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Habitats
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Diet
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Locations
      <span
        className="right"
      />
    </p>
  </div>
</div>
`;

exports[`Test for the details Page Check whether the details page renders correctly 1`] = `
<div
  className="Details"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-arrow-left"
    />
    Home
  </button>
  <div
    className="animalDetails"
  >
    <h2 />
    <img
      alt="undefined details"
    />
    <h2
      className="characteristics"
    >
      Characteristics
    </h2>
    <p
      className="padded"
    >
      Latin Name
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Animal Type
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Active Times
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Minimum Length
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Maximum Length
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Minimum Weight
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Maximum Weight
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Life span
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Habitats
      <span
        className="right"
      />
    </p>
    <p
      className="padded dark"
    >
      Diet
      <span
        className="right"
      />
    </p>
    <p
      className="padded"
    >
      Locations
      <span
        className="right"
      />
    </p>
  </div>
</div>
`;
